Remove the notion of legacy from the api and save some FLASH space too

davids5 opened the pull request:
@LorenzMeier 

This remove the notion of legacy from the api. The board level code  will perform the translation to legacy format on the STM32.  New targets will not need to do this as there is no case where the serial number were used by mfg for tracking.

It maintains the legacy incorrect selection of significant digits of the UUID in the PX4 code base.
This is done to avoid the ripple effects changing the IDs used on STM32 base platforms. [See] (https://github.com/PX4/Firmware/commit/f126668dbd41f10667a8759be7937dd822549828#diff-8da0cce1d272772f52e0d92c6f4343a5R83)

davids5 commented:
Jenkins test this please.

LorenzMeier commented:
From your comment I gather you only kept the display of legacy. If that conclusion would be correct: However, the byte order is also inverted in storage IIRC and via MAVLink. Please compare the UUID as sent via MAVLInk in the MAVLink inspector of QGroundControl. It should stay the same for STM32 as it will blow up existing configs.

If you kept the representation for STM32 then this looks like its ready to go.

davids5 commented:
Jenkins test this please.

davids5 commented:
1.5.0 Release RC1
```
nsh> ver all
HW arch: PX4FMU_V2
FW git-hash: 3d7906647e5c6f382bf6c574927bac88d752874f
FW version: v1.4.4rc1 (1.4.4 192), 17040576
OS version: 6.27 (4294967295)
Build datetime: Aug 26 2016 09:41:20
Toolchain: 4.8.4 20140725 (release) [ARM/embedded-4_8-branch revision 213147]
MCU: STM32F42x, rev. 3
UID: 400038:33355110:32383336
nsh> fmu id
ERROR [fmu] Board serial:
 00400038 33355110 32383336
``` 
![image](https://cloud.githubusercontent.com/assets/1945821/22397688/5f2446ac-e51b-11e6-8cac-51c180adbf5a.png)


Current master
```
nsh> ver all
HW arch: PX4FMU_V2
FW git-hash: 264589b2cc8bc69e95fc63580e2ad94be109c016
FW version: 1.5.2 0 (17105408)
OS: NuttX
OS version: Release 1.8.0 (17301759)
Build datetime: Jan 28 2017 05:19:48
Build uri: localhost
Toolchain: GNU GCC, 5.4.1 20160609 (release) [ARM/embedded-5-branch revision 237715]
MCU: STM32F42x, rev. 3
UID: 400038:33355110:32383336
nsh> fmu id
Board serial:
 400038 33355110 32383336
nsh>
```
![image](https://cloud.githubusercontent.com/assets/1945821/22397624/de360eaa-e519-11e6-8403-11dae9b17aa9.png)

This Branch 
```
nsh> ver all
HW arch: PX4FMU_V2
FW git-hash: b1b8e9b87a96b0b032237e31bb91f4e182a86d69
FW version: 1.5.2 0 (17105408)
OS: NuttX
OS version: Release 1.8.0 (17301759)
Build datetime: Jan 28 2017 05:49:58
Build uri: localhost
Toolchain: GNU GCC, 5.4.1 20160609 (release) [ARM/embedded-5-branch revision 237715]
MCU: STM32F42x, rev. 3
UID: 400038:33355110:32383336
nsh> fmu id
Board serial:
 00400038 33355110 32383336
nsh> WARN  [commander] Not ready to fly: Sensors not set up correctly
```
![image](https://cloud.githubusercontent.com/assets/1945821/22397641/540e77de-e51a-11e6-9ecb-b1a1212d516e.png)

davids5 commented:
@LorenzMeier - Please DNM yet - I have more commits to add here....

davids5 commented:
This branch post discussion
```
nsh> ver all
HW arch: PX4FMU_V2
FW git-hash: 256851bc27a9445e5fb24e5a0ec3516cc34d37d6
FW version: 1.5.2 0 (17105408)
OS: NuttX
OS version: Release 1.8.0 (17301759)
Build datetime: Jan 28 2017 08:20:45
Build uri: localhost
Toolchain: GNU GCC, 5.4.1 20160609 (release) [ARM/embedded-5-branch revision 237715]
MFGUID: 323833363335511000400038
MCU: STM32F42x, rev. 3
UID: 400038:33355110:32383336
nsh> fmu id
FMU: unrecognized command id, try:
  mode_gpio, mode_pwm, mode_pwm4, test, sensor_reset [milliseconds], i2c <bus> <hz>, bind
```
![image](https://cloud.githubusercontent.com/assets/1945821/22398795/42b2acda-e533-11e6-84a6-95b5e49f5882.png)

davids5 commented:
@LorenzMeier - ready!

LorenzMeier commented:
CI catched some code issues:

```
/Users/travis/build/PX4/Firmware/src/systemcmds/ver/ver.c:60:19: fatal error: variable 'mfg_uid_str' is not needed and will not be emitted [-Wunneeded-internal-declaration]
static const char mfg_uid_str[]         = "mfguid";
                  ^
1 error generated.
```

davids5 commented:
@LorenzMeier 

Now that we added the mfguid and display in in ver  I cleaned up the uavcan usage and removed all the exposed reordering from the API and include files. The stm32 board code only needs to do the maintain the legacy format.

I have tested this on px4fmu-v3 and a new 2.4.6 board w/ rev 3 silicon as well as the esc35-v1 hw bootloader.

Node id allocation is AOK with these changes.

```
nsh>  ver all
HW arch: PX4FMU_V2
FW git-hash: 3fe5c4f2372298bf1c44fa655e344d632a31afcd
FW version: 1.6.0 0 (17170432)
OS: NuttX
OS version: Release 1.8.0 (17301759)
Build datetime: Jan 30 2017 13:23:35
Build uri: localhost
Toolchain: GNU GCC, 5.4.1 20160609 (release) [ARM/embedded-5-branch revision 237715]
MFGUID: 323833363335511000400038
MCU: STM32F42x, rev. 3
UID: 400038:33355110:32383336
nsh> ver uid
UID: 400038:33355110:32383336
nsh> ver mfguid
MFGUID: 323833363335511000400038
```
![image](https://cloud.githubusercontent.com/assets/1945821/22447871/5bc01a56-e6f9-11e6-8d15-1301f29926ad.png)

davids5 commented:
@LorenzMeier Please give this a second look.

davids5 commented:
@LorenzMeier Architectural question. Do we want to pad out a 96 bit mfg to 128 bit with leading 0's.

The we can use 32 character string in the logging and add the to QGC as an mfguid[32]?

LorenzMeier commented:
Rebased and force-pushed.

LorenzMeier commented:
@davids5 Padding is fine.

LorenzMeier commented:
I still will merge this once it passes.
